src/reconstruction/ppmc_cuda.cu

```diff
@@ -1,1277 +1,681 @@
 /*! \file ppmc_cuda.cu
  *  \brief Functions definitions for the ppm kernels, using characteristic
  tracing. Written following Stone et al. 2008. */
-#ifdef CUDA
-  #ifdef PPMC
 
-    #include <math.h>
+#include <math.h>
 
-    #include "../global/global.h"
-    #include "../global/global_cuda.h"
-    #include "../reconstruction/ppmc_cuda.h"
-    #include "../utils/gpu.hpp"
-    #include "../utils/hydro_utilities.h"
+#include "../global/global.h"
+#include "../global/global_cuda.h"
+#include "../reconstruction/ppmc_cuda.h"
+#include "../reconstruction/reconstruction.h"
+#include "../utils/gpu.hpp"
+#include "../utils/hydro_utilities.h"
 
-    #ifdef DE  // PRESSURE_DE
-      #include "../utils/hydro_utilities.h"
-    #endif
+#ifdef DE  // PRESSURE_DE
+  #include "../utils/hydro_utilities.h"
+#endif
 
-/*! \fn void PPMC_cuda(Real *dev_conserved, Real *dev_bounds_L, Real
- *dev_bounds_R, int nx, int ny, int nz, int n_ghost, Real dx, Real dt, Real
- gamma, int dir, int n_fields)
+// =====================================================================================================================
+/*!
  *  \brief When passed a stencil of conserved variables, returns the left and
  right boundary values for the interface calculated using ppm. */
-__global__ void PPMC_cuda(Real *dev_conserved, Real *dev_bounds_L, Real *dev_bounds_R, int nx, int ny, int nz,
-                          int n_ghost, Real dx, Real dt, Real gamma, int dir, int n_fields)
+__global__ void PPMC_CTU(Real *dev_conserved, Real *dev_bounds_L, Real *dev_bounds_R, int nx, int ny, int nz, Real dx,
+                         Real dt, Real gamma, int dir)
 {
-  int n_cells = nx * ny * nz;
-  int o1, o2, o3;
-  switch (dir) {
-    case 0:
-      o1 = 1;
-      o2 = 2;
-      o3 = 3;
-      break;
-    case 1:
-      o1 = 2;
-      o2 = 3;
-      o3 = 1;
-      break;
-    case 2:
-      o1 = 3;
-      o2 = 1;
-      o3 = 2;
-      break;
+  // get a thread ID
+  int const thread_id = threadIdx.x + blockIdx.x * blockDim.x;
+  int xid, yid, zid;
+  cuda_utilities::compute3DIndices(thread_id, nx, ny, xid, yid, zid);
+
+  // Thread guard to prevent overrun
```

Comment:
Can you make this comment clearer? I.e. it used to say something like "Check to make sure we are only operating on real cells" or something like that, now it is a sort of meaningless phrase.